test/decoder_test.exs

```diff
@@ -226,16 +234,31 @@ defmodule DecoderTest do
     stream = Stream.map(["a,\"be", "c,d", "e,f\"", "g,h", "i,j", "k,l"], &(&1))
 
     assert_raise SyntaxError, fn ->
-      Decoder.decode(stream, multiline_escape: false) |> Stream.run
+      Decoder.decode!(stream, multiline_escape: false) |> Stream.run
     end
+
+    assert stream
+    |> Decoder.decode(multiline_escape: false)
+    |> Stream.filter(fn
+      { :error, SyntaxError, _, _ } -> true
+      _ -> false
+    end)
+    |> Enum.count == 2
   end
 
   test "raises an error if rows are of variable length" do
     stream = Stream.map(["a,\"be\"", ",c,d", "e,f", "g,,h", "i,j", "k,l"], &(&1))
 
     assert_raise RowLengthError, fn ->
-      Decoder.decode(stream) |> Stream.run
+      Decoder.decode!(stream) |> Stream.run
     end
+
+    assert stream
+    |> Decoder.decode
+    |> Enum.any?(fn
+      { :error, RowLengthError, _, _ } -> true
+    _ -> false
+    end)
```

Comment:
Same here - I am happy to keep the tests for `decode` to cases where it differs from `decode!` for now btw